unity-client/Assets/Scripts/MainScripts/DCL/Controllers/Scene/SceneController.cs

```diff
@@ -225,125 +225,142 @@ public bool ProcessMessage(MessagingBus.QueuedSceneMessage_Scene msgObject, out
+                        break;
+                    }
 
                     case MessagingTypes.ENTITY_DESTROY:
-                        {
-                            if (msgObject.payload is Protocol.RemoveEntity payload)
-                                scene.RemoveEntity(payload.entityId);
-                            break;
-                        }
+                    {
+                        if (msgPayload is Protocol.RemoveEntity payload)
+                            scene.RemoveEntity(payload.entityId);
+                        break;
+                    }
 
                     case MessagingTypes.INIT_DONE:
-                        {
-                            scene.sceneLifecycleHandler.SetInitMessagesDone();
-                            break;
-                        }
+                    {
+                        scene.sceneLifecycleHandler.SetInitMessagesDone();
+                        break;
+                    }
 
                     case MessagingTypes.QUERY:
-                        {
-                            if (msgObject.payload is QueryMessage queryMessage)
-                                ParseQuery(queryMessage.payload, scene.sceneData.id);
-                            break;
-                        }
+                    {
+                        if (msgPayload is QueryMessage queryMessage)
+                            ParseQuery(queryMessage.payload, scene.sceneData.id);
+                        break;
+                    }
 
                     case MessagingTypes.OPEN_EXTERNAL_URL:
-                        {
-                            if (msgObject.payload is Protocol.OpenExternalUrl payload)
-                                OnOpenExternalUrlRequest?.Invoke(scene, payload.url);
-                            break;
-                        }
+                    {
+                        if (msgPayload is Protocol.OpenExternalUrl payload)
+                            OnOpenExternalUrlRequest?.Invoke(scene, payload.url);
+                        break;
+                    }
 
                     case MessagingTypes.OPEN_NFT_DIALOG:
-                        {
-                            if (msgObject.payload is Protocol.OpenNftDialog payload)
-                                OnOpenNFTDialogRequest?.Invoke(payload.contactAddress, payload.tokenId, payload.comment);
-                            break;
-                        }
+                    {
+                        if (msgPayload is Protocol.OpenNftDialog payload)
+                            OnOpenNFTDialogRequest?.Invoke(payload.contactAddress, payload.tokenId,
+                                payload.comment);
+                        break;
+                    }
 
                     default:
-                        Debug.LogError($"Unknown method {method}");
-                        return true;
+                        throw new Exception($"Unknown method.");
```

Comment:
I wouldn't throw an exception here. If we are testing an outdated unity build against a new kernel build that implements a new type of message, it's more dev-friendly if we just fail gracefully and output the error in console. (already had this use case with the builder in world and it helped).

Reply:
👍 done